recirq/qcqmc/trial_wf.py

```diff
@@ -212,6 +213,97 @@ def _json_dict_(self):
         simple_dict["params"] = self.params
         return simple_dict
 
+    @classmethod
+    def build_pp_plus_trial_from_dependencies(
+        cls,
+        params: PerfectPairingPlusTrialWavefunctionParams,
+        *,
+        dependencies: Dict[data.Params, data.Data],
+        do_print: bool = False,
+    ) -> "TrialWavefunctionData":
+        """Builds a TrialWavefunctionData from a TrialWavefunctionParams
+
+        Args:
+            params: The parameters specifying the PP+ trial wavefunction.
+            dependencies: Data dependencies
+            do_print: Print debugging information to stdout
+
+        Returns:
+            The constructed TrialWavefunctionData object.
+        """
+
+        # Avoid circular imports
+        from recirq.qcqmc import optimize_wf
+
+        if do_print:
+            print("Building Trial Wavefunction")
+        np.random.seed(params.seed)
+        hamiltonian_data = dependencies[params.hamiltonian_params]
+        assert isinstance(hamiltonian_data, hamiltonian.HamiltonianData)
+
+        if params.n_orb != params.n_elec:
+            raise ValueError("PP wavefunction must have n_orb = n_elec")
+
+        if params.do_optimization:
+            (
+                one_body_params,
+                two_body_params,
+                one_body_basis_change_mat,
+            ) = optimize_wf.get_pp_plus_params(
+                hamiltonian_data=hamiltonian_data,
+                restricted=params.restricted,
+                random_parameter_scale=params.random_parameter_scale,
+                initial_orbital_rotation=params.initial_orbital_rotation,
+                heuristic_layers=params.heuristic_layers,
+                do_pp=params.do_pp,
+                n_optimization_restarts=params.n_optimization_restarts,
+                do_print=do_print,
+                use_fast_gradients=params.use_fast_gradients,
+            )
+        else:
+            if (
+                params.initial_two_body_qchem_amplitudes is None
+                or params.initial_orbital_rotation is not None
+            ):
+                raise NotImplementedError(
+                    "TODO: Implement whatever isn't finished here."
```

Comment:
What?  Let's at least put a better message here.

Reply:
I also got confused by the logic reading it again. I think I simplified the error message.